cpp/include/cuml/cuml.hpp

```diff
@@ -95,6 +95,9 @@ class cumlHandle {
   * @returns num of streams in the handle.
   */
   int getNumInternalStreams();
+
+  std::vector<cudaStream_t> cumlHandle::getInternalStreams() const;
```

Comment:
I think the interface would probably be more future proof and easier to integrate with non C++ if it would expose `getInternalStream(int stream_id)` returning `cudaStream_t`

Reply:
Thejaswi pointed out moving this function into the handle as we have an increasing number of single- and multi-gpu primitives which are using these internal streams as input and creating arrays out of them by looping through the `getInternalStream(int stream_id)` is causing quite a bit of duplication.

To be more c-friendly, what do you think of exposing a function to return them as an array? `getWorkerStreams(cudaStream_t streams *);`

Reply:
OK I thought this might simplify things on the python side. Making a C++ more C friendly does not make much sense. IMHO. Then ignore my comment.